core/src/main/java/com/turn/ttorrent/client/SharedTorrent.java

```diff
@@ -685,6 +680,11 @@ public synchronized void handlePeerReady(SharingPeer peer) {
 				"that was already requested from another peer.");
 		}
 
+		SortedSet<Piece> rarest = new TreeSet<Piece>();
```

Comment:
Not a fan of the field hiding. Could you name the sorted set `rarestPieces` instead?

Reply:
Sorry, i did not notice. Fixed